lib/streamlit/elements/metric.py

```diff
@@ -0,0 +1,138 @@
+from .utils import clean_text
+
+
+class MetricMixin:
+    def metric(self, label, value, delta=None, delta_colors="normal"):
+        """Display a metric widget.
+
+        Parameters
+        ----------
+        label : str
+            The header or Title for the metric
+        value : Integer, Float, String, or None
+             Value of the metric. None is rendered as a long dash.
+        delta : Integer, Float, String, or None
+            Indicator of how the metric changed, rendered with an arrow below
+            the metric. If delta is negative (int/float) or starts with a minus
+            sign (str), the arrow points down and the text is red; else the
+            arrow points up and the text is green. If None (default), no delta
+            indicator is shown.
+        delta_colors :
+             If "normal" (default), the delta indicator is shown as described
+             above. If "inverse", it is red when positive and green when
+             negative. This is useful when a negative change is considered
+             good, e.g. if cost decreased. If "off", delta is  shown in gray
+             regardless of its value.
+
+        Returns
+        -------
+
+        Example
+        -------
+        >>> st.metric(label="Active developers", value=123, delta=123,
+        ...     delta_colors="off")  # arrow up, gray
+
+        """
+        metric_proto = MetricProto()
+        metric_proto.body = self.parse_value(value)
+        metric_proto.label = self.parse_label(label)
+        metric_proto.delta = self.parse_delta(delta)
+        metric_proto.delta_colors = self.determine_delta_colors(
+            clean_text(delta_colors), delta
+        )
+        return self.dg._enqueue("metric", metric_proto)
+
+    def parse_label(self, label):
+        if not isinstance(label, str):
+            raise TypeError(
+                "'" + str(label) + "' is not an accepted type. label only accepts: str"
+            )
+        return label
+
+    def parse_value(self, value):
+        if value is None:
+            return "—"
+        if isinstance(value, float) or isinstance(value, int) or isinstance(value, str):
+            return str(value)
+        else:
+            raise TypeError(
+                "'" + str(value) + "' is not an accepted type. value only accepts: "
+                "int, float, str, or None"
+            )
+
+    def parse_delta(self, delta):
+        if delta is None:
+            return ""
+        if isinstance(delta, str):
+            delta = dedent(delta)
+            if delta[0] == "-":
+                return delta[1:]
+            return delta
+        elif isinstance(delta, int):
+            return str(abs(delta))
+        elif isinstance(delta, float):
+            return str(abs(delta))
+        else:
+            raise TypeError(
+                "'" + str(delta) + "' is not an accepted type. delta only accepts:"
+                " int, float, str, or None"
+            )
```

Comment:
nit: @willhuang1997 For error string construction you can use `f-strings` or string `.format()` method instead of string concatenation. 

We don't have any special recommendations for that in our [Style Guide ](https://github.com/streamlit/streamlit/wiki/Style-Guide), but I think in that case it will look a little bit more readable. 

```
raise TypeError(
    f"'{str(delta)}' is not an accepted type."
    "Delta only accepts: int, float, str, or None"
)
```